src/datachain/toolkit/split.py

```diff
@@ -58,10 +58,14 @@ def train_test_split(dc: DataChain, weights: list[float]) -> list[DataChain]:
 
     weights_normalized = [weight / sum(weights) for weight in weights]
 
+    resolution = 2**31 - 1  # Maximum positive value for a 32-bit signed integer.
```

Comment:
Ideally, we could use `2**64 - 1` and get rid of the `%` operation, but @dmpetrov expressed concerns at https://github.com/iterative/datachain/pull/604#discussion_r1848903487 about differences between the different database implementations.

```suggestion
    resolution = 2**64 - 1  # Maximum positive value for a 64-bit unsigned integer.
```